test/scripts/update-param-sys.py

```diff
@@ -0,0 +1,97 @@
+cmd("Linux_Example Command_Router_Instance-Reset_Data_Products")
+wait(1.0)
+
+# Send Noop_Arg command expecting success:
+cmd("Linux_Example Command_Router_Instance-Noop_Arg with Value 1")
+wait(1.0)
+# Check successful command count is 2 and that the last success was Noop_Arg with last Value 1:
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Success_Count.Value == '2'")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Successful_Command.Id == '3'")
+print("Noop_Arg success OK")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Noop_Arg_Last_Value.Value == '1'")
+print("Noop_Arg last Value OK")
+
+# Send Noop command expecting failure:
+cmd("Linux_Example Command_Router_Instance-Noop_Arg with Value 868")
+wait(1.0)
+# Check last failed command count was 1 and that it was Noop_Arg with last Value 1:
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Failure_Count.Value == '1'")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Failed_Command.ID == '3'")
+print("Noop_Arg failure OK")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Noop_Arg_Last_Value.Value == '868'")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Failed_Command.Status == 'FAILURE'")
+print("Noop_Arg last Value failure OK")
+
+#########################################################################
+# If this point was reached then begin testing parameter update functions
+#########################################################################
+
+# Build a parameter table using bytearrays:
+# Osc_A_Freq
+Test_Packed_Table = bytearray(struct.pack(">f", 0.30))
+# Osc_A_Amp
+Test_Packed_Table += bytearray(struct.pack(">f", 5.00))
+# Osc_A_Off
+Test_Packed_Table += bytearray(struct.pack(">f", 2.50))
+# Osc_B_Freq
+Test_Packed_Table += bytearray(struct.pack(">f", 0.30))
+# Osc_B_Amp
+Test_Packed_Table += bytearray(struct.pack(">f", -5.00))
+# Osc_B_Off
+Test_Packed_Table += bytearray(struct.pack(">f", -2.50))
+# Version
+Test_Packed_Table += bytearray(struct.pack(">f", 0.0))
+# Crc_Table
+Test_Packed_Table += bytearray(struct.pack(">h", 0))
+# Buffer_Length
+Test_Packed_Table += bytearray(struct.pack(">h", 55))
+
+# Send nominal Update_Parameter_Table command expecting success:
+cmd("Linux_Example", "Parameter_Manager_Instance-Update_Parameter_Table", {
+      "Header.Table_Buffer_Length": 24,
+      "Header.Crc_Table": 19692,
+      "Table_Buffer": list(Test_Packed_Table)
+   })
+wait(1.0)
+# Check successful command count is 3 and that it was Update_Parameter_Table:
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Success_Count.Value == '3'")
+check_formatted("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Successful_Command.Id == '5200'")
+print("Update_Parameter_Table success OK")
```

Comment:
You should send a command to dump the table here, and should verify that the dumped table (and CRC) matches what is expected.

Reply:
Added in [fad3e09](https://github.com/lasp/adamant_example/pull/33/commits/fad3e091fcf57aee94a97e13e40e3a8d008b398c). Currently checking this is hardcoded using `wait_check` on each value in `Parameters_Instance.Active_Parameters`, but `get_tlm` in the Script Runner API could be used in the future (this returns a "packet hash" and would need to be parsed.)

Reply:
[44c9542](https://github.com/lasp/adamant_example/pull/33/commits/44c954293916fa7d33ca0e7be6b9ae85019a21ba) implements directly checking Active_Parameter CRC and table buffer bytes against the initial bytearrays created for the test.